packages/common-all/src/types/ReducedDEngine.ts

```diff
@@ -5,6 +5,7 @@ import { DEngine } from "./typesv2";
  */
 export type ReducedDEngine = Pick<
   DEngine,
+  | "wsRoot"
```

Comment:
I don't think we should expose these kinds of properties anymore - engine interfaces should be focused on note/schema CRUD only, and any access to other things should be provided elsewhere.  This will help us avoid big monolithic objects like `DExtension`.

Reply:
so i needed to expose this because we need `wsRoot` to compare vaults for equality (sometimes vaults can have absolute file paths and so we need the wsRoot to compare them). we can mark it as protected if we don't want others to use them